unified-runtime/source/adapters/level_zero/v2/queue_batched.hpp

```diff
@@ -0,0 +1,562 @@
+//===--------------- queue_batched.hpp - Level Zero Adapter ---------------===//
+//
+// Copyright (C) 2025 Intel Corporation
+//
+// Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM
+// Exceptions. See LICENSE.TXT
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "../common.hpp"
+#include "../device.hpp"
+
+#include "command_list_cache.hpp"
+#include "common/ur_ref_count.hpp"
+#include "context.hpp"
+#include "event.hpp"
+#include "event_pool_cache.hpp"
+#include "memory.hpp"
+#include "queue_api.hpp"
+
+#include "ur/ur.hpp"
+
+#include "command_buffer.hpp"
+#include "command_list_manager.hpp"
+#include "lockable.hpp"
+#include "queue_immediate_in_order.hpp"
+#include "ur_api.h"
+#include "ze_api.h"
+
+namespace v2 {
+
+struct batch_manager {
+private:
+  ur_command_list_manager activeBatch;
+  ur_command_list_manager immediateList;
+  std::vector<v2::raii::command_list_unique_handle> runBatches;
+  ur_event_generation_t regularGenerationNumber;
+  static constexpr uint64_t default_num_batches = 10;
+
+public:
+  batch_manager(ur_context_handle_t context, ur_device_handle_t device,
+                v2::raii::command_list_unique_handle &&commandListRegular,
+                v2::raii::command_list_unique_handle &&commandListImmediate)
+      : activeBatch(context, device,
+                    std::forward<v2::raii::command_list_unique_handle>(
+                        commandListRegular)),
+        immediateList(context, device,
+                      std::forward<v2::raii::command_list_unique_handle>(
+                          commandListImmediate)),
+        regularGenerationNumber(0) {
+    runBatches.reserve(default_num_batches);
+  }
+
+  friend class ur_queue_batched_t;
```

Comment:
why do you need this? It'd be better if you could use only the public interfaces from the batched queue. Having to use internals sort of leaks abstraction (especially since the architecture is: queue -> batch manager -> batch).